analysis/src/machine_check.rs

```diff
@@ -333,6 +333,12 @@ impl TypeChecker {
                                 Item::TypeDeclaration(enum_decl),
                             );
                         }
+                        asm::SymbolValue::TraitDeclaration(trait_decl) => {
```

Comment:
I was just wondering if we should combine TypeDeclaration and TraitDeclaration, but I realized that traits are not types, they cannot be used where types are expected - if you want to do that, you have to introduce a type var and bind it to the trait. Rust does it the same, it just allows traits to be used as a shortcut for the "correct" way.